lib/permissions.js

```diff
@@ -529,16 +523,57 @@ exports = module.exports = {
             }
           }
 
-          roles.forEach(function (r) {
-            r.statement.forEach(function (s, index) {
-              s.resource = s.resource.replace('{{tenantid}}', tenantId);
-              statements.push(s);
-            });
-          });
+          if (!roles || 0 === roles.length) {
+            return callback(null, []);
+          }
 
-          return callback(null, statements);
-        });
-      });
+          async.each(
+            roles,
+            function (r, nextRole) {
+              async.each(
+                r.statement,
+                function (s, nextStatement) {
+                  s.resource = s.resource.replace('{{tenantid}}', tenantId);
+                  statements.push(s);
+                  nextStatement();
+                },
+                nextRole
+              );
+            },
+            function (err) {
+              if (err) {
+                return callback(err);
+              }
+
+              // next retrieve bare policies
+              db.retrieve('policy', {user: userId}, function (error, records) {
+                if (err) {
+                  return callback(err);
+                }
+
+                if (!records || 0 === records.length) {
```

Comment:
I don't think there is a preferred format as such.  Some developers like to write it like this as it can sometimes accidental assignment when you really meant to compare values.